vsphere/datadog_checks/vsphere/api.py

```diff
@@ -91,26 +107,29 @@ def smart_connect(self):
             err_msg = "Connection to {} failed: {}".format(self.config.hostname, e)
             raise APIConnectionError(err_msg)
 
-        if self._conn:
+        if getattr(self, '_conn', None):
             connect.Disconnect(self._conn)
-        self._conn = conn
+
+        self._conn = conn  # type: ServiceInstance
```

Comment:
I understand the intent of not wanting to deal with `Optional[ServiceInstance]`, but I think it might be cleaner to keep the "default to `None` on init" and add explicit checks on `self._conn` in methods that try to access it, eg:

```python
@smart_retry
def check_health(self):
    if self._conn is None:
        raise RuntimeError('Not connected')
    self._conn.CurrentTime()
```

Reply:
Hum. `self._conn.` is used in around 7+ methods, so, that might lead to a lot of boiler plate.

Reply:
Use another approach, a bit cleaner: `self._conn = cast(ServiceInstance, None)`